feat: 유저 단건 조회 기능

jinseong205 opened the pull request:
유저 조회 dto getUserResponse 작성
controller, service 빈 등록 및 기능 구현
repository jpaRepository 상속 및 기능 구현
유저 조회 단위 테스트
resolved #8

jinseong205 committed:
feat: 유저 단건 조회 기능

유저 조회 dto getUserResponse 작성
controller, service 빈 등록 및 기능 구현
repository jpaRepository 상속 및 기능 구현
유저 조회 단위 테스트

jinseong205 committed:
feat: 유저 단건 조회 confilct 해결

유저 단건 조회 confilct 해결
유저 조회 url 변경

daengdaengLee commented on src/main/java/hello/until/user/dto/response/getUserResponse.java:
파일명이랑 클래스명 대문자로 시작해야 할 것 같습니다.

daengdaengLee commented on src/main/java/hello/until/user/dto/response/getUserResponse.java:
```diff
@@ -0,0 +1,21 @@
+
+@Getter
+public class getUserResponse {
+    private Long id;
+    private String email;
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
```
final 처리해도 될 것 같아요.

daengdaengLee commented on src/main/java/hello/until/user/service/UserService.java:
```diff
@@ -36,6 +37,11 @@ private void validateUser(String email) {
+    @Transactional(readOnly = true)
+    public User getUserById(Long id){
+        return userRepository.findById(id)
+                .orElseThrow(()-> new EntityNotFoundException());
```
조회 기능 같은 경우 서비스 레이어까진 Optional 로 리턴하고
컨트롤러 레이어에서 예외 발생시키는 게 어떨까요?

daengdaengLee commented on src/test/java/hello/until/user/service/UserServiceTest.java:
없는 회원 조회일 경우 테스트도 필요할 것 같아요.

daengdaengLee commented on src/main/java/hello/until/user/controller/UserController.java:
```diff
@@ -17,12 +21,18 @@
 	
-}
\ No newline at end of file
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getPostById(@PathVariable Long id){
+        User user = userService.getUserById(id);
+        return ResponseEntity.ok(new getUserResponse(user));
+    }
```
회원 못 찾은 경우 처리가 빠진 것 같습니다.

daengdaengLee reviewed (commented):
LGTM!
코멘트 확인 부탁드려요!

jinseong205 committed:
feat: review 내용 적용

review 내용 적용하여 수정
- 미회원 가입자 테스트 추가
- 회원 조회 오류처리 service -> controller 단으로 변경
- 회원 조회 반환값 Optional<User> 처리

daengdaengLee commented on src/main/java/hello/until/user/controller/UserController.java:
```diff
@@ -17,12 +22,18 @@
\ No newline at end of file
+    @GetMapping("/{id}")
+    public ResponseEntity<?> getUserById(@PathVariable long id){
+        User user = userService.getUserById(id).orElseThrow(() -> new EntityNotFoundException("해당 유저를 찾을 수 없습니다."));
```
EntityNotFoundException 는 영속성 레이어 예외라서, Http 응답에 맞는 예외를 쓰는 게 좋을 것 같습니다!
ErrorResponseException 이거에 404 코드 넣어서 쓰는 게 어떨까요?

jinseong205 committed:
feat: 컨트롤러 오류처리 수정

컨트롤러 오류 처리 수정하여 업로드
